### DISCUSSION TITLE
[ZK Stack] - It is taking more than 10 hours to deploy ZK Stack on sepolia

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#692] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [x] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Im trying to deploy ZK chain with sepolia as base L1. It has been more than 10 hours and still it has been not deployed yet. Im using custom token.

![Web_Photo_Editor](https://github.com/user-attachments/assets/ec827157-b915-41b0-853f-ef5b9d4bd4be)

### Expected Behavior

Ideally whole thing should be deployed within 2 hours including build time.

### Current Behavior

Very slow deployment of smart contracts on L1 sepolia.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears that you are experiencing an unusually long deployment time for the ZK Stack on the Sepolia network. This issue might be influenced by several factors, including network congestion on Sepolia, the complexity of the custom token you are deploying, or potential misconfigurations in your deployment setup.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Network Status**: Verify the current status of the Sepolia network to ensure there are no ongoing issues that could be affecting deployment times.

2. **Review Deployment Scripts**: Ensure that your deployment scripts are optimized and correctly configured. Double-check the gas settings and related parameters.

3. **Logs and Outputs**: Review the logs generated during the deployment process for any errors or warnings that could provide more insight into what might be causing the delay.

4. **Simplify the Deployment**: If possible, try deploying a simpler version of your smart contracts to determine if the issue is related to the specific complexities of your custom token.

5. **Community and Support**: Consider reaching out on the [ZKsync developers GitHub discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for community support or insights from others who might have faced similar issues.

If these steps do not resolve the issue, it may be beneficial to provide more detailed information about your deployment configuration and any error logs you have, which could help in diagnosing the problem more effectively.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
Are you following any other guides or tutorials? That will be helpful to reproduce what you're getting.

**Reply by sachanayush47** [reply]
> Are you following any other guides or tutorials? That will be helpful to reproduce what you're getting.

Yeah, Im following only these 2 tutorials:
- https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain
- https://docs.zksync.io/zk-stack/running-a-zk-chain/quickstart

**Reply by sachanayush47** [reply, marked as answer]
Hi, It has been deployed without taking much time in 3rd attempt.
